src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -0,0 +1,107 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Client
+{
+    // Todo need to be async
+
+    public class OrasClient : IOrasClient
+    {
+        private readonly string _imageReference;
+
+        public OrasClient(string imageReference)
+        {
+            EnsureArg.IsNotNull(imageReference, nameof(imageReference));
+
+            _imageReference = imageReference;
+        }
+
+        public async Task<bool> PullImageAsync(string outputFolder)
+        {
+            string command = $"pull  {_imageReference} -o {outputFolder}";
```

Comment:
Do we need to check the existence of output folder here?

Reply:
If output folder not exist, oras will create the folder.
If exist, files will be override. 
We can avoid override outside the function, let user use an empty folder.

Reply:
do we need check empty or strange inputs that could make oras crash?

Reply:
Directly use oras exception?